vxquery-core/src/main/java/org/apache/vxquery/hdfs2/HDFSFunctions.java

```diff
@@ -0,0 +1,192 @@
+
+            conf.addResource(new Path(this.conf_path + "/core-site.xml"));
+            conf.addResource(new Path(this.conf_path + "/hdfs-site.xml"));
+            try {
+                fs = FileSystem.get(conf);
+            } catch (IOException ex) {
+                System.err.println(ex);
+            }
+        } else {
+            System.err.println("Could not locate hdfs configuarion folder.");
+        }
+    }
+
+    /**
+     * Returns true if the file path exists or it is located somewhere in the home directory of the user that called the function.
+     * Searches in subdirectories of the home directory too.
+     * 
+     * @param filename
+     * @return
+     */
+    public boolean isLocatedInHDFS(String filename) {
+        try {
+            //search file path
+            if (fs.exists(new Path(filename))) {
+                return true;
+            }
+        } catch (IOException ex) {
+            System.err.println(ex);
+        }
+        //Search every file and folder in the home directory
+        if (searchInDirectory(fs.getHomeDirectory(), filename) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Searches the given directory and subdirectories for the file.
+     * 
+     * @param directory
+     *            to search
+     * @param filename
+     *            of file we want
+     * @return path if file exists in this directory.else return null.
+     */
+    public Path searchInDirectory(Path directory, String filename) {
+        //Search every folder in the directory
+        try {
+            RemoteIterator<LocatedFileStatus> it = fs.listFiles(directory, true);
+            String[] parts;
+            Path path;
+            while (it.hasNext()) {
+                path = it.next().getPath();
+                parts = path.toString().split("/");
+                if (parts[parts.length - 1].equals(filename)) {
+                    return path;
+                }
+            }
+        } catch (IOException ex) {
+            System.err.println(ex);
+        }
+        return null;
+    }
+
+    /**
+     * Read the cluster properties file and locate the HDFS_CONF variable that is the directory path for the
+     * hdfs configuration if the system environment variable HDFS_CONF is not set.
+     * 
+     * @return true if is successfully finds the Hadoop/HDFS home directory
+     */
+    private boolean locateConf() {
+
+        this.conf_path = System.getProperty("HDFS_CONF");
+        if (this.conf_path == null) {
+
+            // load properties file
+            Properties prop = new Properties();
+            String propFilePath = "../vxquery-server/src/main/resources/conf/cluster.properties";
+            try {
+                prop.load(new FileInputStream(propFilePath));
+            } catch (FileNotFoundException e) {
+                propFilePath = "vxquery-server/src/main/resources/conf/cluster.properties";
+                try {
```

Comment:
When I ran the project from eclipse it uses as current working directory the path vxquery/vxquery-cli so in order to get the properties file with a relative path I use the first defined propFilePath.When I ran the project from terminal my working directory is vxquery so the second relative propFilePath works.I am looking for a better solution to this static definition of the propFilePath.